src/webrtc/datachannel.ts

```diff
@@ -285,28 +285,25 @@ export class DataChannelClass implements DataChannel {
   }
 
   // Assumes data is chunked.
-  private handleSendDataToPeer_ = (data:Data) : Promise<void> => {
+  private handleSendDataToPeer_ = (data:Data) : void => {
     try {
       if(typeof data.str === 'string') {
-        this.rtcDataChannel_.send(data.str);
+        this.rtcDataChannel_.send.reckless(data.str);
       } else if(data.buffer) {
         this.toPeerDataBytes_ -= data.buffer.byteLength;
-        this.rtcDataChannel_.sendBuffer(data.buffer);
+        this.rtcDataChannel_.sendBuffer.reckless(data.buffer);
         this.lastBrowserBufferedAmount_ += data.buffer.byteLength;
       } else {
         // Data is good when it meets the type expected of the Data. If type-
         // saftey is ensured at compile time, this should never happen.
-        return Promise.reject(new Error(
-            'Bad data: ' + JSON.stringify(data)));
+        throw new Error('Bad data: ' + JSON.stringify(data));
       }
     // Can raise NetworkError if channel died, for example.
     } catch (e) {
```

Comment:
i just noticed this nasty "log and re-throw" anti-pattern -- might be an opportunity now to remove it?